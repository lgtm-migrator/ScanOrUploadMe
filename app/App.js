import React from 'react';
import { Platform, StatusBar, StyleSheet, View } from 'react-native';
import { AppLoading } from 'expo';
import { Asset } from 'expo-asset';
import * as Icon from '@expo/vector-icons';
import * as Font from 'expo-font';

import TabNavigator from './src/navigation/TabNavigator';

// Redux
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import reducers from './src/reducers/index';

// global store
const store = createStore(reducers);

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoadingComplete: false
    };
  }

  _loadResourcesAsync = async () => {
    return Promise.all([
      Asset.loadAsync([
        require('./assets/images/robot-dev.png'),
        require('./assets/images/robot-prod.png')
      ]),
      Font.loadAsync({
        // tab bar font
        ...Icon.Ionicons.font
      })
    ]);
  };

  _handleLoadingError = error => {
    // can be used to report the error to the error reporting service, for example Sentry
    console.warn(error);
  };

  _handleFinishLoading = () => {
    this.setState({ isLoadingComplete: true });
  };

  render() {
    if (!this.state.isLoadingComplete && !this.props.skipLoadingScreen) {
      return (
        <AppLoading
          startAsync={this._loadResourcesAsync}
          onError={this._handleLoadingError}
          onFinish={this._handleFinishLoading}
        />
      );
    } else {
      return (
        <View style={styles.container}>
          <Provider store={store}>
            {Platform.OS === 'ios' && <StatusBar barStyle="default" />}
            <TabNavigator />
          </Provider>
        </View>
      );
    }
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff',
    flex: 1
  }
});
