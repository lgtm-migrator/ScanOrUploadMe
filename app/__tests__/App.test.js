import "react-native";
import React from "react";
import App from "../App";
import renderer from "react-test-renderer";
import NavigationTestUtils from "react-navigation/NavigationTestUtils";

describe("App Snapshot", () => {
  jest.useFakeTimers();

  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it("Renders the loading screen", async () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders the root without loading screen", async () => {
    const tree = renderer.create(<App skipLoadingScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("two plus two is four", () => {
    expect(2 + 2).toBe(4);
  });

  it("works", () => {
    expect(1).toBe(1);
  });
});
