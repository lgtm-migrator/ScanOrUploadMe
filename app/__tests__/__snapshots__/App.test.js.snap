// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot Renders the loading screen 1`] = `null`;

exports[`App Snapshot Renders the root without loading screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "opacity": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": 50,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onGestureEvent={[Function]}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#eee",
                    "flex": 1,
                    "flexDirection": "column-reverse",
                    "overflow": "hidden",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    pointerEvents="auto"
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "paddingTop": 64,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                        ],
                      }
                    }
                  >
                    <View
                      accessibilityElementsHidden={false}
                      style={
                        Object {
                          "backgroundColor": "#fff",
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "column",
                            "justifyContent": "flex-start",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "marginHorizontal": 10,
                              "marginTop": 10,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "rgba(96,100,109, 1)",
                                "fontSize": 16,
                                "lineHeight": 24,
                                "textAlign": "center",
                              }
                            }
                          >
                            Choose how you are going to provide the input image.
                          </Text>
                        </View>
                        <View
                          pointerEvents="box-none"
                          style={
                            Array [
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "bottom": 0,
                                  "left": 0,
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                },
                                Object {
                                  "elevation": undefined,
                                  "justifyContent": "flex-end",
                                  "zIndex": undefined,
                                },
                              ],
                              undefined,
                            ]
                          }
                        >
                          <View
                            pointerEvents="none"
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "bottom": 0,
                                "elevation": undefined,
                                "justifyContent": "flex-end",
                                "left": 0,
                                "opacity": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                                "zIndex": undefined,
                              }
                            }
                          />
                          <View
                            pointerEvents="box-none"
                            style={
                              Array [
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "bottom": 0,
                                    "left": 0,
                                    "position": "absolute",
                                    "right": 0,
                                    "top": 0,
                                  },
                                  Object {
                                    "elevation": undefined,
                                    "justifyContent": "flex-end",
                                    "zIndex": undefined,
                                  },
                                ],
                                Object {
                                  "alignItems": "flex-end",
                                },
                                Object {
                                  "paddingVertical": 30,
                                },
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "marginHorizontal": 30,
                                    "zIndex": undefined,
                                  },
                                  Object {
                                    "elevation": 5,
                                    "shadowColor": "#000",
                                    "shadowOffset": Object {
                                      "height": 5,
                                      "width": 0,
                                    },
                                    "shadowOpacity": 0.35,
                                    "shadowRadius": 3,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <View
                                accessible={true}
                                isTVSelectable={true}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "opacity": 1,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "rgba(231, 76, 60, 1)",
                                      "borderRadius": 28,
                                      "height": 56,
                                      "width": 56,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "borderRadius": 28,
                                        "height": 56,
                                        "justifyContent": "center",
                                        "transform": Array [
                                          Object {
                                            "scale": 1,
                                          },
                                          Object {
                                            "rotate": "0deg",
                                          },
                                        ],
                                        "width": 56,
                                      }
                                    }
                                  >
                                    <Text
                                      style={
                                        Object {
                                          "backgroundColor": "transparent",
                                          "color": "rgba(255, 255, 255, 1)",
                                          "fontSize": 24,
                                          "marginTop": -4,
                                        }
                                      }
                                    >
                                      +
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  onLayout={[Function]}
                  pointerEvents="box-none"
                  style={
                    Object {
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#FFF",
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                        ],
                      }
                    }
                  >
                    <View
                      onLayout={[Function]}
                      pointerEvents="box-none"
                      style={
                        Object {
                          "backgroundColor": "#FFF",
                          "borderBottomColor": "#A7A7AA",
                          "borderBottomWidth": 0.5,
                          "height": 64,
                          "paddingBottom": 0,
                          "paddingLeft": 0,
                          "paddingRight": 0,
                          "paddingTop": 20,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "bottom": 0,
                              "flexDirection": "row",
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        >
                          <View
                            pointerEvents="box-none"
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "transparent",
                                "bottom": 0,
                                "flexDirection": "row",
                                "justifyContent": "center",
                                "left": 0,
                                "opacity": 1,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                          >
                            <Text
                              accessibilityTraits="header"
                              accessible={false}
                              allowFontScaling={true}
                              numberOfLines={1}
                              onLayout={[Function]}
                              style={
                                Object {
                                  "color": "rgba(0, 0, 0, .9)",
                                  "fontSize": 17,
                                  "fontWeight": "600",
                                  "marginHorizontal": 16,
                                  "textAlign": "center",
                                }
                              }
                            >
                              ScanMe!
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "backgroundColor": "#fff",
          "borderTopColor": "rgba(0, 0, 0, .3)",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "height": 49,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        accessibilityLabel="Home, tab, 1 of 3"
        accessible={true}
        hitSlop={
          Object {
            "bottom": 5,
            "left": 15,
            "right": 15,
            "top": 5,
          }
        }
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 29,
              },
              false,
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 1,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text />
          </View>
        </View>
        <Text
          allowFontScaling={true}
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "lightblue",
              "fontSize": 11,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          Home
        </Text>
      </View>
      <View
        accessibilityLabel="Saved, tab, 2 of 3"
        accessible={true}
        hitSlop={
          Object {
            "bottom": 5,
            "left": 15,
            "right": 15,
            "top": 5,
          }
        }
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 29,
              },
              false,
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 1,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text />
          </View>
        </View>
        <Text
          allowFontScaling={true}
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "gray",
              "fontSize": 11,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          Saved
        </Text>
      </View>
      <View
        accessibilityLabel="Settings, tab, 3 of 3"
        accessible={true}
        hitSlop={
          Object {
            "bottom": 5,
            "left": 15,
            "right": 15,
            "top": 5,
          }
        }
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 29,
              },
              false,
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 1,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text />
          </View>
        </View>
        <Text
          allowFontScaling={true}
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "gray",
              "fontSize": 11,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          Settings
        </Text>
      </View>
    </View>
  </View>
</View>
`;
